### PR TITLE
892 minor api changes

### LINKED ISSUE #892: Minor api changes
* Add otherPartyName to InvoiceFilterInput
* Change nameId on InvoiceFilterInput to otherPartyId
* Add type to ItemNode and ItemFilter
* Add item resolver to StockLine
* Normalise tax input to be of TaxUpdateType
* Add requisition sort on comment

### PR DESCRIPTION
**Pull request by andreievg**
closes #892

changes as per commits

### CODE REVIEW

File: graphql/invoice_line/src/mutations/inbound_shipment_line/service_line/insert.rs [hunk 1]
```diff
@@ -89,7 +90,7 @@ impl InsertInput {
             name,
             total_before_tax,
             total_after_tax,
-            tax,
+            tax: tax.map(|tax| tax.percentage).flatten(),
```
Comment (clemens-msupply) [inline graphql/invoice_line/src/mutations/inbound_shipment_line/service_line/insert.rs:93]: could be done with `tax.and_then(|tax| tax.percentage)`

File: graphql/invoice_line/src/mutations/outbound_shipment_line/line/insert.rs [hunk 2]
```diff
@@ -97,7 +98,7 @@ impl InsertInput {
             number_of_packs,
             total_before_tax,
             total_after_tax,
-            tax,
+            tax: tax.map(|tax| tax.percentage).flatten(),
```
Comment (clemens-msupply) [inline graphql/invoice_line/src/mutations/outbound_shipment_line/line/insert.rs:101]: and_then

File: graphql/invoice_line/src/mutations/outbound_shipment_line/service_line/insert.rs [hunk 3]
```diff
@@ -89,7 +90,7 @@ impl InsertInput {
             name,
             total_before_tax,
             total_after_tax,
-            tax,
+            tax: tax.map(|tax| tax.percentage).flatten(),
```
Comment (clemens-msupply) [inline graphql/invoice_line/src/mutations/outbound_shipment_line/service_line/insert.rs:93]: and_then

**Review by clemens-msupply — APPROVED** [review]

File: graphql/invoice_line/src/mutations/inbound_shipment_line/service_line/insert.rs (hunk 1 above)
Comment (andreievg) [reply graphql/invoice_line/src/mutations/inbound_shipment_line/service_line/insert.rs]: Thanks, 6c8e1e215dfca7c155aa4bb8e1ca730a68f7a08b, this commit also replaces all flatten(), including sort mapping and key_value_store

**Pull request merged**